compiler/src/dotty/tools/dotc/tastyreflect/TypeOrBoundsTreesOpsImpl.scala

```diff
@@ -19,6 +19,10 @@ trait TypeOrBoundsTreesOpsImpl extends scala.tasty.reflect.TypeOrBoundsTreeOps w
   def TypeTreeDeco(tpt: TypeTree): TypeTreeAPI = new TypeTreeAPI {
     def pos(implicit ctx: Context): Position = tpt.pos
     def tpe(implicit ctx: Context): Type = tpt.tpe.stripTypeVar
+    def symbol(implicit ctx: Context): Symbol = {
+      val sym = tpt.symbol
+      if (sym.isType) sym else sym.companionClass
```

Comment:
A document is required for the behavior of `symbol`? What may happen is `sym` is a `TermRef` but not an object.

Reply:
I changed the logic to work as it does in the compiler. Now `symbol` just returns the symbol. Then one must use `companionClass` from the TASTy reflect interface.